packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -196,9 +195,20 @@ class _FlutterValidator extends DoctorValidator {
       'engine revision ${version.engineRevisionShort}'
     ));
 
+    String dartVersion = _getDartVersion();
+    if (dartVersion != null)
+      messages.add(new ValidationMessage(dartVersion));
+    else
+      messages.add(new ValidationMessage.error('Unable to find the Dart executable on the path.'));
+
     return new ValidationResult(ValidationType.installed, messages,
       statusInfo: 'on ${osName()}, channel ${version.channel}');
   }
+
+  String _getDartVersion() {
+    ProcessResult result = Process.runSync('dart', <String>['--version']);
+    return result.exitCode != 0 ? null : result.stderr.trim();
+  }
```

Comment:
This is confusing because its the version of Dart that the flutter tool is using, which might not match the version of Dart that the Flutter app is running.